articles/tuples-data-structure-python/index.md

```diff
@@ -0,0 +1,307 @@
+- [Further Reading](#further-reading)
+
+### Tuples: Let's Code
+
+As we discussed, a Tuple is a collection of items that are immutable. Let us start by creating a tuple.
+
+#### Creating a Tuple
+
+A tuple can be created in multiple ways. The simplest way of creating a tuple is by setting a variable to a pair of empty parantheses.
+
+```py
+# Set the tuple1 variable to an empty tuple
+tuple1 = ()
+
+# Check if the tuple is initialized properly
+print(type(tuple1))
+
+# Output : <class 'tuple'>
+```
+
+The above code snippet gives an output of `<class: 'tuple'>`, which indicates that the tuple has been created successfully. We can also create a tuple by using the in-built `tuple()` method in Python.
+
+```py
+# Set the tuple2 variable to an empty tuple by using the tuple() method
+tuple2 = tuple()
+
+# Check if the tuple is initialized properly
+print(type(tuple2))
+
+# Output : <class 'tuple'>
+```
+
+While initializing a tuple, we can also specify what data exists inside it.
+
+```py
+# tuple3 consists of values 40, 50, 60
+tuple3 = (40, 50, 60)
+
+# tuples can also consist of different datatypes
+tuple4 = (45, "55", "Hello World", True, 42.6)
+
+# we can also create a tuple of lists
+tuple5 = ([10, 20], [30, 40], [50, 60])
+```
+
+#### Accessing Items in a Tuple
+
+Tuples follow [zero indexing](https://en.wikipedia.org/wiki/Zero-based_numbering). In zero indexing, the first element of the tuple has the index '0', the second element of the tuple has the index '1', and so on.
+
+##### Positive Indexing
+
+For example, let us create a tuple, `tuple1`. Tuple elements can be accessed the same way as a list element.
+
+```py
+tuple1 = (0, 1, 2, 3)
+
+print(tuple1[0]) # Output: 0
+
+print(tuple1[1]) # Output: 1
+```
+
+This tuple follows zero indexing.
+
+![Tuple Positive Indexing](/engineering-education/tuples-data-structure-python/tuple-positive-index.png)
+
+_Tuple Positive Indexing: Source -- [GeeksforGeeks](https://www.geeksforgeeks.org/python-tuples/)_
+
+##### Negative Indexing
+
+Similar to lists, we can also use negative indexing on a tuple. Therefore, '-1' refers to the Nth element of a tuple, -2 refers to the (N-1)th element, and so on (where N is the length of the tuple).
+
+```py
+tuple1 = (30, 40, 50, 60)
+
+print(tuple1[-1]) # Output: 60
+
+print(tuple1[-3]) # Output: 40
+```
+
+![Tuple Negative Indexing](/engineering-education/tuples-data-structure-python/tuple-negative-index.png)
+
+_Tuple Negative Indexing_
+
+##### Slicing
+
+In Python, slicing is used to return a range of values. Like lists, tuples can also be sliced.
+
+```py
+tuple1 = (1, 2, 3, 5, 8, 13)
+
+print(tuple1[0:3]) # Output: (1, 2, 3)
+
+print(tuple1[4:]) # Output: (8, 13)
+```
+
+As per the examples shown above, if we slice a range of [a : b), it would return from tuple index a to tuple index (b - 1). For more tricks on python slicing, check out [this page](https://stackoverflow.com/questions/509211/understanding-slice-notation).
```

Comment:
I think you will have to use $ sign to specify the range. Else, this appears to be broken. $ [a,b) $. At the end, Math JAX file will have to be inserted. If you find a work around, please proceed. \\[a,b\\). This works.